### PR TITLE
3670 Fix channel_dim in ITKReader and add in it NibabelReader

### LINKED ISSUE #3670: Nibabel reader need to support channel first data
**Is your feature request related to a problem? Please describe.**
Currently, our `ITKReader` can specify the `channel_dim` to read channel first file:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/data/image_reader.py#L171
But `NibabelReader` doesn't support it.
Would be nice to also support `channel_dim` in `NibabelReader`.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #3670 .

### Description
This PR fixed channel_dim in ITKReader and added in it NibabelReader.
Still under development, just a proposal PR for discussion.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: tests/test_load_imaged.py [hunk 1]
```diff
@@ -63,10 +63,12 @@ def test_channel_dim(self):
         test_image = np.random.rand(*spatial_size)
         with tempfile.TemporaryDirectory() as tempdir:
             filename = os.path.join(tempdir, "test_image.nii.gz")
-            nib.save(nib.Nifti1Image(test_image, affine=np.eye(4)), filename)
+            itk_np_view = itk.image_view_from_array(test_image.T.copy())
```
Comment (wyli) [inline tests/test_load_imaged.py:66]: shall we keep the pervious implementation, it's a valid test case.

**Review by wyli — APPROVED** [review]
Thanks, it looks good to me apart from the extra test_load_imaged change

File: tests/test_load_imaged.py (hunk 1 above)
Comment (Nic-Ma) [reply tests/test_load_imaged.py]: Sure, restored the test in the latest commit.

Thanks.

**Pull request merged**